internal/plugin/connectors/tcp/mssql/connection_details.go

```diff
@@ -69,15 +74,18 @@ func (cd *ConnectionDetails) Address() string {
 	return cd.Host + ":" + strconv.FormatUint(uint64(cd.Port), 10)
 }
 
-func newSSLOptions(credentials map[string][]byte) map[string]string {
-	SSLOptions := make(map[string]string)
+func newSSLParams(credentials map[string][]byte) map[string]string {
+	sslMode := string(credentials["sslmode"])
+	params, ok := sslModeToBaseParams[sslMode]
+
+	if !ok {
+		credentials["sslmode"] = defaultSSLMode
+		return newSSLParams(credentials)
+	}
 
-	for _, sslOption := range sslOptions {
-		value := string(credentials[sslOption])
-		if len(value) > 0 {
-			SSLOptions[sslOption] = value
-		}
+	if sslMode == "verify-ca" {
+		params["rawcertificate"] = string(credentials["sslrootcert"])
```

Comment:
Let's confirm that passing in a raw certificate is URL-safe, though I suspect not. This might mean that we should explore using a driver constructor that takes in the params map instead of the DSN string.